src/main/java/com/incentives/piggyback/user/controller/UserController.java

```diff
@@ -25,14 +29,25 @@
     @Autowired
     private UserService userService;
 
+    @Autowired
+    private UserEventPublisher.PubsubOutboundGateway messagingGateway;
+
     @PostMapping("/user")
     public ResponseEntity<User> createUser(@Valid @RequestBody User user) {
         log.debug("User Service: Received POST request for creating new user.");
-        //check no conflict
         User newUser = userService.save(user);
         URI location = ServletUriComponentsBuilder.fromCurrentRequest()
                 .query(newUser.getId().toString())
                 .buildAndExpand(newUser.getId().toString()).toUri();
+        //PUSHING MESSAGES TO GCP
+        messagingGateway.sendToPubsub(
+                CommonUtility.stringifyEventForPublish(
+                        newUser.toString(),
+                        Constant.USER_CREATED_EVENT,
+                        Calendar.getInstance().getTime().toString(),
+                        HttpStatus.CREATED.toString(),
```

Comment:
4th event field is meant for specifying partnerId, so that an event can be mapped to a partner - for billing and invoicing. For events not related to partners directly, it should be passed as empty.